src/SimulationMap.cu

```diff
@@ -1,13 +1,241 @@
 #include "SimulationMap.cuh"
 #include "Polyhedron.cuh"
-#include "common.cuh"
+#include "Face.cuh"
+#include "jones_constants.hpp"
+#include "random_generator.cuh"
+#include <cmath>
 
+namespace jc = jones_constants;
 
-// Implement SimulationMap::SimulationMap(...) here
+
+__device__ double const mapnode_dist = 2 * jc::speed;
+
+
+__device__ SimulationMap::SimulationMap(Polyhedron *polyhedron) :
```

Comment:
At the moment this code is hard to read. I'm pretty sure it can be simplified by removing some duplications and/or splitting into smaller methods, or at least I'd appreciate some comments inside of the ctor's body